lib/libc/stdlib/malloc/snmalloc/Makefile.inc

```diff
@@ -0,0 +1,48 @@
+.PATH: ${LIBC_SRCTOP}/stdlib/malloc/snmalloc
+
+MISRCS+=	malloc.cc
+
+# Specify the locations in contrib for headers.
+CXXFLAGS+=	-I${SRCTOP}/contrib/subrepo-snmalloc/src/snmalloc
+
+CXXSTD?=	c++20
+
+# C++ dialect options.
+CXXFLAGS+=	-fno-exceptions -fno-rtti
+.if ${MACHINE_ARCH} == amd64
+# -mcx16 is needed to avoid calls to libatomic.
+CXXFLAGS+=	-mcx16
+.endif
+
+# Require an explicit call from libthr to clean up thread-local state
+CXXFLAGS+=	-DSNMALLOC_USE_THREAD_CLEANUP
+# Export the __je_bootstrap_* functions for TLS allocation.
+CXXFLAGS+=	-DSNMALLOC_BOOTSTRAP_ALLOCATOR
+# Export compatibility implementations of the jemalloc non-standard and
+# experimental APIs that are now part of the ABI contract for libc.
+CXXFLAGS+=	-DSNMALLOC_JEMALLOC3_EXPERIMENTAL
+CXXFLAGS+=	-DSNMALLOC_JEMALLOC_NONSTANDARD
+# Use getentropy for random numbers.
+CXXFLAGS+=	-DSNMALLOC_PLATFORM_HAS_GETENTROPY
+# Prefix all of these functions with double underscore:
+CXXFLAGS+=	-DSNMALLOC_STATIC_LIBRARY_PREFIX=__
+
+.if ${MK_MALLOC_PRODUCTION} != "yes" || !defined(MALLOC_PRODUCTION)
+CXXFLAGS+=	-DSNMALLOC_CHECK_CLIENT
+CXXFLAGS+=	-DSNMALLOC_FAIL_FAST=false
+.endif
+
+# Don't emit frame pointers, the fast path for malloc is sufficiently small
+# that this adds measurable overhead.
+SHARED_CXXFLAGS+=	-fomit-frame-pointer
+STATIC_CXXFLAGS+=	-fomit-frame-pointer
+PO_CXXFLAGS+=	-fno-omit-frame-pointer
```

Comment:
This seems dubious to me; surely when profiling you want the same code to be generated?..